etc/telegraf.conf

```diff
@@ -1123,6 +1123,123 @@
 #   ## Export metric collection time.
 #   # export_timestamp = false
 
+## Configuration for apache pulsar  to send metrics to
+#[[outputs.pulsar]]
+# ## URLs of pulsar url
+#  url = "pulsar://localhost:6650"
+# ## Pulsar topic for producer messages
+#  topic = "persistent://public/default/telegraf"
+#
+# ## The value of this tag will be used as the topic.  If not set the 'topic'
+# ## option is used.
+# # topic_tag = "foo"
+#
+# ## If true, the 'topic_tag' will be removed from to the metric.
+# # exclude_topic_tag = false
+#
+#  routing_tag = "host"
+#
+# ## The routing key is set as the message key and used to determine which
+# ## partition to send the message to.  This value is only used when no
+# ## routing_tag is set or as a fallback when the tag specified in routing tag
+# ## is not found.
+# ##
+# ## If set to "random", a random value will be generated for each message.
+# ##
+# ## When unset, no message key is added and each message is routed to a random
+# ## partition.
+# ##
+# ##   ex: routing_key = "random"
+# ##       routing_key = "telegraf"
+# # routing_key = ""
+#
+# ## Optional Authentication Provider Config Defaults to empty "" NoAuthentication
+# ## if set to "token" provide the JWT token
+# ## if set to "tls" the please mention tls_certificate_path and tls_private_key_path
+# # auth_provider = ""
+#
+## For token auth provider
+# # auth_token = ""
+# # Set the following values for tls auth provider
+# # tls_allow_insecure_connection = false
+# # tls_trust_certs_file_path = ""
+# # tls_validate_host_name = true
+# # tls_certificate_path = ""
+# # tls_private_key_path = ""
+#
+# ## Optional timeout Config
+# # connection_timeout = "30s"
+# # operation_timeout = "30s"
+#
+# ## Optional Producer Config
+#
+# ## CompressionType represents the various compression codecs recognized by
+# ## Pulsar in messages.
+# ##  0 : No compression
+# ##  1 : LZ4 compression
+# ##  2 : ZLib compression
+# ##  3 : ZSTD compression
+# # compression_type = 0
+#
+# ## MaxPendingMessages set the max size of the queue holding the messages pending to receive an
+# ## acknowledgment from the broker.
+# # max_pending_messages = 1000
+#
+# ## BatchingMaxPublishDelay set the time period within which the messages sent will be batched (default: 10ms)
+# ## if batch messages are enabled. If set to a non zero value, messages will be queued until this time
+# ## interval or until
+# # batching_max_publish_delay = "10ms"
+#
+# ## BatchingMaxMessages set the maximum number of messages permitted in a batch. (default: 1000)
+# ## If set to a value greater than 1, messages will be queued until this threshold is reached or
+# ## batch interval has elapsed.
+# # batching_max_messages = 1000
```

Comment:
Can we set this to the metric_batch_size?

Reply:
@danielnelson I am using the length of the []metrics to determine the batch size. Is it possible to directly get metric_batch_size defined in the config? This will help me to completely discard this option.   Because length may not tell the batch size. and batch size is set at producer creation times in pulsar unlike kafka where it is set and every sendMessages.

Reply:
@danielnelson is it possible to get the metric_batch_size config? If so how can I get it?  Right now I am using length of []metrics to determine the batch size. In pulsar, batch size needs to be set at Producer creation time unlike in kafka where batch size can be adjusted everytime  we send messages.